framework/src/auxkernels/NodalPatchRecoveryAuxBase.C

```diff
@@ -24,54 +21,49 @@ NodalPatchRecoveryAux::validParams()
   return params;
 }
 
-NodalPatchRecoveryAux::NodalPatchRecoveryAux(const InputParameters & parameters)
-  : AuxKernel(parameters), _npr(getUserObject<NodalPatchRecoveryBase>("nodal_patch_recovery_uo"))
+NodalPatchRecoveryAuxBase::NodalPatchRecoveryAuxBase(const InputParameters & parameters)
+  : AuxKernel(parameters)
 {
-  // Check user object block restriction for consistency
-  if (!isBlockSubset(_npr.blockIDs()))
-    paramError("nodal_patch_recovery_uo",
-               "Nodal patch recovery auxiliary kernel is not defined in a subset of blocks of the "
-               "associateduser object. Revise your input file.");
+  if (!isNodal())
+    mooseError(name(), " only runs on nodal variables.");
 }
 
 Real
-NodalPatchRecoveryAux::computeValue()
+NodalPatchRecoveryAuxBase::computeValue()
 {
-  if (!isNodal())
-    mooseError(name(), " only runs on nodal variables.");
-
   // get node-to-conneted-elem map
   const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map = _mesh.nodeToElemMap();
   auto node_to_elem_pair = node_to_elem_map.find(_current_node->id());
   mooseAssert(node_to_elem_pair != node_to_elem_map.end(), "Missing entry in node to elem map");
```

Comment:
We could also do a `libmesh_map_find` here and then do a structured binding for more readability

Reply:
Maybe you don't use `first` ... in which case I often do `libmesh_ignore(foo)`

Reply:
Oh never mind ... I see you re-assign the iterator later